homeassistant/components/ruckus_unleashed/config_flow.py

```diff
@@ -28,34 +41,38 @@ async def validate_input(hass: core.HomeAssistant, data):
     """
 
     try:
-        ruckus = await Ruckus.create(
+        async with AjaxSession.async_create(
             data[CONF_HOST], data[CONF_USERNAME], data[CONF_PASSWORD]
-        )
-    except AuthenticationError as error:
-        raise InvalidAuth from error
-    except ConnectionError as error:
-        raise CannotConnect from error
-
-    mesh_name = await ruckus.mesh_name()
-
-    system_info = await ruckus.system_info()
-    try:
-        host_serial = system_info[API_SYSTEM_OVERVIEW][API_SERIAL]
-    except KeyError as error:
-        raise CannotConnect from error
-
-    return {
-        "title": mesh_name,
-        "serial": host_serial,
-    }
+        ) as ruckus:
+            system_info = await ruckus.api.get_system_info(
+                SystemStat.SYSINFO,
+                SystemStat.IDENTITY,
+                SystemStat.UNLEASHED_NETWORK,
+            )
+            zd_name = system_info[API_SYS_IDENTITY][API_SYS_IDENTITY_NAME]
+            zd_serial = (
+                system_info[API_SYS_UNLEASHEDNETWORK][API_SYS_UNLEASHEDNETWORK_TOKEN]
+                if API_SYS_UNLEASHEDNETWORK in system_info
+                else system_info[API_SYS_SYSINFO][API_SYS_SYSINFO_SERIAL]
+            )
+            return {
+                KEY_SYS_TITLE: zd_name,
```

Comment:
The mesh_name from the old library is the same as zd_name?

Reply:
No. I didn't think mesh name was a great choice for the data entry flow form title.

mesh_name is the name of the hidden SSID used by APs for wireless backhaul. It's a 17 digit random number, or placeholder text if wireless backhaul hasn't been enabled on a ZoneDirector.

zd_name is the name of your Ruckus Unleashed or ZoneDirector network, chosen when you run through the initial setup wizard (but able to be changed afterwards). It would typically be something boring like "ruckus".

Reply:
reverted